From 2b007826486cb4d02e712789ee72a8175c8fa7bd Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Sat, 16 Oct 2021 10:52:13 +0000
Subject: [PATCH backport 5.10 088/182] dt-bindings: Add description for
 EMC2305 for RPM-based PWM Fan Speed Controller

Michrochip EMC2305 RPM-based PWM Fan Speed Controller.

Signed-off-by: Michael Shych <michaelsh@nvidia.com>
---
 .../devicetree/bindings/hwmon/emc2305.yaml    | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/hwmon/emc2305.yaml

diff --git a/Documentation/devicetree/bindings/hwmon/emc2305.yaml b/Documentation/devicetree/bindings/hwmon/emc2305.yaml
new file mode 100644
index 000000000000..8a46c02a3d83
--- /dev/null
+++ b/Documentation/devicetree/bindings/hwmon/emc2305.yaml
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+
+$id: http://devicetree.org/schemas/hwmon/emc2305.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: RPM-based PWM Fan Speed Controller EMC2305
+
+maintainers:
+  - Claud Chang<claud.chang@deltaww.com>
+
+properties:
+  compatible:
+    enum:
+      - emc2305
+
+  emc2305,min-pwm:
+    description:
+      Min pwm of emc2305
+    maxItems: 1
+  emc2305,max-pwm:
+    description:
+      Max pwm of emc2305
+    maxItems: 1
+  emc2305,cooling-levels:
+    description:
+      Quantity of cooling level state.
+    maxItems: 1
+
+required:
+  - compatible
+  - emc2305,min-pwm
+  - emc2305,max-pwm
+  - emc2305,cooling-levels
+
+additionalProperties: false
+
+examples:
+  - |
+    fan@1 {
+        compatible = "smsc,emc2305";
+        emc2305,min-pwm = <0>;
+        emc2305,max-pwm = <255>;
+        emc2305,cooling-levels = <10>;
+    }
-- 
2.20.1

